envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt

```diff
@@ -73,6 +74,7 @@ class RemoteServices(
             .getState(instance)
             .checkpoint("cross-dc-service-update-$cluster")
             .name("cross-dc-service-update-$cluster").metrics()
+            .map { ServicesState(it.serviceNameToInstances.filterValues { value -> value.instances.isNotEmpty() }) }
```

Comment:
Maybe we should do it also for local services?

Reply:
It might be possible but it might break our `HealthIndicator`. If an instance of `ServicesState` won't change then EC will be unhealthy. So when in Consul we have services without instances then EC won't be healthy.

Reply:
But I think it would be unhealthy only if there are zero services with instances in consul. If there is a service without instances, we can skip it and not save in `ConsulLocalClusterStateChanges.latestServiceState`. If there is at least one service with instances, `latestServiceState` will have instances, so the HC will pass. Correct me if I am wrong.

Anyway, I am not suggesting to do it in this PR. It is more like a discussion for future improvements.